### ISSUE TITLE
Version 14 issues after upgrading to latest version

### ISSUE THREAD

**Issue by ian-sayles** [firebase/FirebaseUI-iOS#1199]
Since the Build issues with 14 have been fixed, which I encountered last week, I have again tried upgrading my project to the latest version of FirebaseUI.

FireabaseUI  13.2.0 to 14.2.1
Firebase SDK 10.29.0 to 11.0.0

My app only uses the Auth functionality of FirebaseUI, Sign in with email, Microsoft, Apple and Google.

Below are all the issues I have encountered, is there any upgrade docs to show what needs doing to current apps before upgrading to fix these issues, or are these bugs in FirebaseUI?

Sign in with email:   

When sign in with account  already created I get after entering the email "This type of account isn't supported by this app"

Create a new account everything works ok.

Sign in with Apple:

I get the following error:
```
*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Call must be made on main thread'
*** First throw call stack:
(0x195758f20 0x18d6532b8 0x194c5f7f8 0x197c51df4 0x197b8e92c 0x197ec3f90 0x197ec3aa0 0x197ec4edc 0x197ec697c 0x197ec09b4 0x197ec06dc 0x197ec0570 0x10585e478 0x10585e13c 0x10585e03c 0x10595b26c 0x105746f38 0x10574d85d 0x105724f81 0x1057250d5 0x1057252f5 0x10572544d 0x1a0aa6775)
libc++abi: terminating due to uncaught exception of type NSException
```

Sign in with Google:
```
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'No active configuration.  Make sure GIDClientID is set in Info.plist.'
*** First throw call stack:
(0x195758f20 0x18d6532b8 0x1958576dc 0x101e33b2c 0x101e32694 0x101cac8b4 0x101e1a3e0 0x101e20efc 0x197d524f8 0x197d523d0 0x197d52220 0x198269200 0x19826aa14 0x1979e1ce0 0x197adcb04 0x197c9374c 0x197c92a10 0x197b13924 0x197b15118 0x197b1de10 0x197a10740 0x197a0e660 0x197a0e2a4 0x197a0f148 0x19572b834 0x19572b7c8 0x195729298 0x195728484 0x195727cd8 0x1da1751a8 0x197d61ae8 0x197e15d98 0x19991c294 0x1998c8860 0x1998d461c 0x10178b33c 0x10178b658 0x1b8eff154)
libc++abi: terminating due to uncaught exception of type NSException
```

Sign in with Microsoft:

After signing in with email and password:
```
Main Thread Checker: UI API called on a background thread: -[UIViewController view]
PID: 528, TID: 28824, Thread name: (none), Queue name: com.apple.root.user-initiated-qos.cooperative, QoS: 25
Backtrace:
4   App                        0x0000000105377310 __71-[FUIOAuth signInWithDefaultValue:presentingViewController:completion:]_block_invoke + 356
5   App                          0x0000000105162ef8 $s12FirebaseAuth13OAuthProviderC10credential4withAA0B10CredentialCAA0B10UIDelegate_pSg_tYaKFyyYacfU_ToTQ0_ + 128
6   libswift_Concurrency.dylib          0x00000001a0a9e764 A0C8B407-0ABF-3C28-A54C-FE8B1D3FA7AC + 317284
7   libswift_Concurrency.dylib          0x00000001a0a9f9c8 A0C8B407-0ABF-3C28-A54C-FE8B1D3FA7AC + 321992
8   libdispatch.dylib                   0x000000010a512ae4 _dispatch_root_queue_drain + 404
9   libdispatch.dylib                   0x000000010a5134d8 _dispatch_worker_thread2 + 188
10  libsystem_pthread.dylib             0x00000001f212d8f8 _pthread_wqthread + 228
11  libsystem_pthread.dylib             0x00000001f212a0cc start_wqthread + 8
Main Thread Checker: UI API called on a background thread: -[UIViewController view]
PID: 528, TID: 28824, Thread name: (none), Queue name: com.apple.root.user-initiated-qos.cooperative, QoS: 25
Backtrace:
4   App                          0x0000000105377310 __71-[FUIOAuth signInWithDefaultValue:presentingViewController:completion:]_block_invoke + 356
5   App                          0x0000000105162ef8 $s12FirebaseAuth13OAuthProviderC10credential4withAA0B10CredentialCAA0B10UIDelegate_pSg_tYaKFyyYacfU_ToTQ0_ + 128
6   libswift_Concurrency.dylib          0x00000001a0a9e764 A0C8B407-0ABF-3C28-A54C-FE8B1D3FA7AC + 317284
7   libswift_Concurrency.dylib          0x00000001a0a9f9c8 A0C8B407-0ABF-3C28-A54C-FE8B1D3FA7AC + 321992
8   libdispatch.dylib                   0x000000010a512ae4 _dispatch_root_queue_drain + 404
9   libdispatch.dylib                   0x000000010a5134d8 _dispatch_worker_thread2 + 188
10  libsystem_pthread.dylib             0x00000001f212d8f8 _pthread_wqthread + 228
11  libsystem_pthread.dylib             0x00000001f212a0cc start_wqthread + 8
=================================================================
Main Thread Checker: UI API called on a background thread: -[UIActivityIndicatorView initWithActivityIndicatorStyle:]
PID: 528, TID: 28824, Thread name: (none), Queue name: com.apple.root.user-initiated-qos.cooperative, QoS: 25
Backtrace:
4   App                          0x00000001052799c4 +[FUIAuthBaseViewController addActivityIndicator:] + 156
5   App                          0x000000010537732c __71-[FUIOAuth signInWithDefaultValue:presentingViewController:completion:]_block_invoke + 384
6   App                          0x0000000105162ef8 $s12FirebaseAuth13OAuthProviderC10credential4withAA0B10CredentialCAA0B10UIDelegate_pSg_tYaKFyyYacfU_ToTQ0_ + 128
7   libswift_Concurrency.dylib          0x00000001a0a9e764 A0C8B407-0ABF-3C28-A54C-FE8B1D3FA7AC + 317284
8   libswift_Concurrency.dylib          0x00000001a0a9f9c8 A0C8B407-0ABF-3C28-A54C-FE8B1D3FA7AC + 321992
9   libdispatch.dylib                   0x000000010a512ae4 _dispatch_root_queue_drain + 404
10  libdispatch.dylib                   0x000000010a5134d8 _dispatch_worker_thread2 + 188
11  libsystem_pthread.dylib             0x00000001f212d8f8 _pthread_wqthread + 228
12  libsystem_pthread.dylib             0x00000001f212a0cc start_wqthread + 8
Main Thread Checker: UI API called on a background thread: -[UIActivityIndicatorView initWithActivityIndicatorStyle:]
PID: 528, TID: 28824, Thread name: (none), Queue name: com.apple.root.user-initiated-qos.cooperative, QoS: 25
Backtrace:
4   App                          0x00000001052799c4 +[FUIAuthBaseViewController addActivityIndicator:] + 156
5   App                          0x000000010537732c __71-[FUIOAuth signInWithDefaultValue:presentingViewController:completion:]_block_invoke + 384
6   App                          0x0000000105162ef8 $s12FirebaseAuth13OAuthProviderC10credential4withAA0B10CredentialCAA0B10UIDelegate_pSg_tYaKFyyYacfU_ToTQ0_ + 128
7   libswift_Concurrency.dylib          0x00000001a0a9e764 A0C8B407-0ABF-3C28-A54C-FE8B1D3FA7AC + 317284
8   libswift_Concurrency.dylib          0x00000001a0a9f9c8 A0C8B407-0ABF-3C28-A54C-FE8B1D3FA7AC + 321992
9   libdispatch.dylib                   0x000000010a512ae4 _dispatch_root_queue_drain + 404
10  libdispatch.dylib                   0x000000010a5134d8 _dispatch_worker_thread2 + 188
11  libsystem_pthread.dylib             0x00000001f212d8f8 _pthread_wqthread + 228
12  libsystem_pthread.dylib             0x00000001f212a0cc start_wqthread + 8
-[UIImageView _invalidateImageLayouts] must be called on the main queue
nw_read_request_report [C5] Receive failed with error "Socket is not connected"
nw_read_request_report [C5] Receive failed with error "Socket is not connected"
nw_read_request_report [C5] Receive failed with error "Socket is not connected"
Task <E59C393D-D300-4BDC-96D6-93146BD8F3AC>.<1> HTTP load failed, 189/0 bytes (error code: -1005 [4:-4])
Task <E59C393D-D300-4BDC-96D6-93146BD8F3AC>.<1> finished with error [-1005] Error Domain=NSURLErrorDomain Code=-1005 "The network connection was lost." UserInfo={_kCFStreamErrorCodeKey=-4, NSUnderlyingError=0x30221b3f0 {Error Domain=kCFErrorDomainCFNetwork Code=-1005 "(null)" UserInfo={NSErrorPeerAddressKey=<CFData 0x300fb3340 [0x1fcf43c70]>{length = 16, capacity = 16, bytes = 0x100201bb8efab3ea0000000000000000}, _kCFStreamErrorCodeKey=-4, _kCFStreamErrorDomainKey=4}}, _NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <E59C393D-D300-4BDC-96D6-93146BD8F3AC>.<1>, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDataTask <E59C393D-D300-4BDC-96D6-93146BD8F3AC>.<1>"
), NSLocalizedDescription=The network connection was lost., NSErrorFailingURLStringKey=https://firebaselogging-pa.googleapis.com/v1/firelog/legacy/batchlog, NSErrorFailingURLKey=https://firebaselogging-pa.googleapis.com/v1/firelog/legacy/batchlog, _kCFStreamErrorDomainKey=4}
*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Modifications to the layout engine must not be performed from a background thread after it has been accessed from the main thread.'
*** First throw call stack:
(0x195758f20 0x18d6532b8 0x1b69fc224 0x1b69fbee4 0x1979f0f50 0x197971e20 0x197971914 0x1979747cc 0x19797329c 0x105279d14 0x10537732c 0x105162ef8 0x10516985d 0x105140f81 0x1051410d5 0x1051412f5 0x10514144d 0x1a0aa6775)
libc++abi: terminating due to uncaught exception of type NSException
```

**Comment by ncooke3** [comment]
Thanks for the report! It looks like some UI code needs to get dispatched to the main queue.

**Comment by paulb777** [comment]
After initial investigation, this looks like a FirebaseAuth regression. We should have a fix in 11.1 or 11.2. In the meantime, if you're seeing this, I recommend you stay on Firebase 10.x.

**Comment by paulb777** [comment]
The GoogleSignIn issue is fixed in the released FirebaseUI 14.2.2.

The MainThread issues are fixed in Firebase 11.1.0, which we plan to publish the week of August 19th.

If you'd like to try in advance, update your installation with this one line: https://github.com/firebase/firebase-ios-sdk/pull/13505/files#diff-1efe87bca97c0a4390c4c1880a4689ce9f5dba99cdf1effe9f6a6ee02ff10e7bR344 

Thanks again for the report and let us know if you see any subsequent issues.

**Comment by paulb777** [comment]
@ncooke3 noticed that we did not address the Email Auth UI issue. I'll reopen and we'll work on a fix for that in an upcoming FirebaseUI 14.2.3
